scripts/pre_commit_linter.py

```diff
@@ -1927,6 +1933,99 @@ def _check_for_copyright_notice(self):
+        file/dir. Note that this checks the CODEOWNERS file according to the
+        glob patterns supported by Python2.7 environment. For more information
+        please refer https://docs.python.org/2/library/glob.html.
+        """
+        if self.verbose_mode_enabled:
+            print 'Starting CODEOWNERS file check'
+            print '----------------------------------------'
+
+        with _redirect_stdout(_TARGET_STDOUT):
+            codeowner_file = '.github/CODEOWNERS'
+            failed = False
+            summary_messages = []
+            # Checks whether every pattern in the CODEOWNERS file matches at
+            # least one dir/file.
+            path_patterns = []
+            for line_num, line in enumerate(FileCache.readlines(
+                    codeowner_file)):
+                if line.strip() != '' and line.strip()[0] != '#':
+                    if '@' not in line:
+                        print ('%s --> Pattern on line %s doesn\'t have'
+                               'codeowner' % (codeowner_file, line_num))
+                        failed = True
+                    else:
+                        line_in_concern = line.split('@')[0].strip()
+                        if line_in_concern.endswith('/'):
+                            line_in_concern = line_in_concern[:-1]
+                        if not glob.glob(line_in_concern.replace('/', '', 1)):
+                            print ('%s --> Pattern on line %s doesn\'t match '
+                                   'any file or directory' % (
+                                       codeowner_file, line_num))
+                            failed = True
+                        if (not line_in_concern.startswith('/') and
+                                not './' +
+                                line_in_concern in CODEOWNER_FILE_PATHS):
+                            print ('%s --> Pattern on line %s is invalid. Use '
+                                   'full path relative to the root directory'
+                                   % (codeowner_file, line_num))
+                            failed = True
+                        if '**' in line_in_concern:
+                            print ('%s --> Pattern on line %s is invalid. '
+                                   '\'**\' wildcard not allowed' % (
+                                        codeowner_file, line_num))
+                            failed = True
+                        path_patterns.append(line_in_concern.replace(
+                            '/', '', 1))
+
+            # Checks that every dir/file is covered under CODEOWNERS.
+            for root, _, filename in os.walk('.'):
+                for file_name in filename:
+                    is_root_file = False
+                    if os.path.join(root, file_name) in CODEOWNER_FILE_PATHS:
+                        is_root_file = True
+                    if any(root.startswith(
+                            x) for x in CODEOWNER_DIR_PATHS) or is_root_file:
```

Comment:
Do not use variables such as `x`. They do not explain any thing.

Reply:
change the name! PTAL.